### PR TITLE
Enable unified_mode for all resources

### PR DESCRIPTION
**Pull request by detjensrobert**
# Description

Enables `unified_mode` on all resources to prepare for Chef 17.

Also fixes idempotency in the test recipe and enables idempotency checks.

## Check List

- [x] All tests pass. See TESTING.md for details.
- [ ] New functionality includes testing. -- n/a
- [ ] New functionality has been documented in the README if applicable. -- n/a

### CODE REVIEW

**Comment by ramereth** [comment]
@detjensrobert mind adding the missing changelog entries while you're at it please?

File: kitchen.yml [hunk 1]
```diff
@@ -26,18 +27,28 @@ suites:
...
     run_list:
       - recipe[test::ldap]
+      - recipe[test::sleep]
 
   - name: proxy
     run_list:
       - recipe[test::proxy]
+    provisioner:
+      enforce_idempotency: false
+      multiple_converge: 1
```
Comment (damacus) [inline kitchen.yml:44]: Is this resource not idempotent at the moment?

**Comment by damacus** [comment]
@detjensrobert any idea why those test are failing now?

File: kitchen.yml (hunk 1 above)
Comment (detjensrobert) [reply kitchen.yml:44]: The `grafana_user` update is not.

**Comment by detjensrobert** [comment]
Looks like the dashboard API has changed since the test was last written. Dashboards need to be referred to by UID instead their slug/name. Is it worth updating for that since I see there's talk of axing the dashboard manipulation entirely?

Deprecated in 5.0, removed in 8.0
https://grafana.com/docs/grafana/v7.5/http_api/dashboard/#get-dashboard-by-slug

File: kitchen.yml (hunk 1 above)
Comment (ramereth) [reply kitchen.yml:44]: @detjensrobert can you add a comment to the other suites that aren't (likely due to this?)

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,10 +4,16 @@ This file is used to list changes made in each version of grafana.
 
 ## Unreleased
 
+- Enable `unified_mode` for all resources
```
Comment (ramereth) [inline CHANGELOG.md:7]: Can you also add the deprecation here as well so folks know.

File: resources/dashboard.rb [hunk 3]
```diff
@@ -19,6 +19,11 @@
 # limitations under the License.
 #
 
+unified_mode true
+
+deprecated 'The `grafana_dashboard` resources will be removed in the next major version of the grafana cookbook. ' \
+  'Grafana 8.0 removed support for the dashboard API these use. Use e.g. Terraform to manage dashboards instead.'
```
Comment (ramereth) [inline resources/dashboard.rb]: Odd, I'm not seeing this deprecation when I run it.

**Review by ramereth — CHANGES_REQUESTED** [review]
You need to lock installation to grafana 7.x in the tests or remove the tests that use the deprecated resources.

File: resources/dashboard.rb (hunk 3 above)
Comment (bmhughes) [reply resources/dashboard.rb]: I've had a quick poke around and I'm not entirely unconvinced that they aren't broken at the minute, they definitely did use to work. From what I saw the `deprecated` method in `chef_class.rb` (assuming that's the correct one) doesn't seem to be getting hit for whatever reason. I've not had time to dig into it any further yet.

Comment (ramereth) [reply resources/dashboard.rb]: @lamont-granquist :point_up:

**Review by ramereth — DISMISSED** [review]
LGTM, @bmhughes do you have any further input before we merge this?

**Comment by bmhughes** [comment]
Even though it's broken at the minute I'd like to get the deprecation warnings added into the other API resources as well before rolling this out. So `:folder`, `:datasource`, `:organisation`, `:user` (maybe more).

**Review by ramereth — DISMISSED** [review]

File: resources/user.rb [hunk 4]
```diff
@@ -21,6 +21,9 @@
 
 unified_mode true
 
+deprecated 'The `grafana_user` resource will be removed in the next major version of the grafana cookbook. ' \
+  'Use e.g. Terraform to interact with the Grafana API instead.'
```
Comment (xorima) [inline resources/user.rb]: This doesn't read right to me, maybe something like below would be better?
```suggestion
  'Use another tool such as Terraform to interact with the Grafana API instead.'
```

**Review by ramereth — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [9.7.0](https://github.com/sous-chefs/grafana/releases/tag/9.7.0)

**Pull request merged**